src/swap/utils.ts

```diff
@@ -1,243 +1,250 @@
-      ) as PromiseFulfilledResult<SwapQuoteWithPool>[]
+        (result) => result.status === "fulfilled" && result.value !== undefined
+      ) as PromiseFulfilledResult<SwapQuote>[]
     ).map((result) => result.value);
+
+    return filteredResults;
   });
 }
 
 /**
- * Gets quotes for fixed input swap from each pool passed as an argument,
+ * Gets quotes for fixed input swap the pools and swap router,
  * and returns the best quote (with the highest rate).
  */
-export async function getFixedInputSwapQuote({
-  pools,
-  assetIn,
-  assetOut,
-  amount
-}: {
-  pools: {info: V1PoolInfo | V2PoolInfo; reserves: PoolReserves}[];
-  assetIn: Pick<TinymanAnalyticsApiAsset, "id" | "decimals">;
-  assetOut: Pick<TinymanAnalyticsApiAsset, "id" | "decimals">;
-  amount: number | bigint;
-}): Promise<SwapQuoteWithPool> {
-  const quotePromises = pools.map<Promise<SwapQuoteWithPool>>((pool) => {
-    return new Promise((resolve, reject) => {
-      let quote: SwapQuote | undefined;
-
-      const quoteGetterArgs = {
-        pool: pool.info,
-        assetIn: {amount, id: Number(assetIn.id)},
-        decimals: {assetIn: assetIn.decimals, assetOut: assetOut.decimals},
-        reserves: pool.reserves
-      };
-
-      try {
-        if (pool.info.contractVersion === CONTRACT_VERSION.V1_1) {
-          quote = SwapV1_1.getFixedInputSwapQuote(quoteGetterArgs);
-        } else {
-          quote = SwapV2.getFixedInputSwapQuote(quoteGetterArgs);
+export async function getFixedInputSwapQuote(
+  params: GetSwapQuoteBySwapTypeParams
+): Promise<SwapQuote> {
+  const {amount, assetIn, assetOut, isSwapRouterEnabled, pools} = params;
+
+  const quotePromises: Promise<SwapQuote>[] = [];
+
+  const v1_1Pool = pools.find(
+    (pool) => pool.info.contractVersion === CONTRACT_VERSION.V1_1
+  );
+
+  if (v1_1Pool) {
+    const quoteGetterArgs = {
+      pool: v1_1Pool.info,
+      assetIn: {amount, id: Number(assetIn.id)},
+      decimals: {assetIn: assetIn.decimals, assetOut: assetOut.decimals},
+      reserves: v1_1Pool.reserves
+    };
+
+    quotePromises.push(
+      new Promise((resolve, reject) => {
+        try {
+          resolve(SwapV1_1.getFixedInputSwapQuote(quoteGetterArgs));
+        } catch (error) {
+          reject(error);
         }
+      })
+    );
+  } else {
+    quotePromises.push(
+      Promise.reject(
+        new SwapQuoteError(
+          SwapQuoteErrorType.NoAvailablePoolError,
+          "Trying to swap from non-existent pool"
+        )
+      )
+    );
+  }
 
-        resolve({pool, quote});
-      } catch (error) {
-        reject(error);
-      }
-    });
-  });
+  const v2Pool = pools.find((pool) => pool.info.contractVersion === CONTRACT_VERSION.V2);
+
+  quotePromises.push(
+    SwapV2.getFixedInputSwapQuote({
+      amount,
+      assetInID: getAssetId(assetIn),
+      assetOutID: getAssetId(assetOut),
+      pool: v2Pool?.info ?? null,
+      decimals: {assetIn: assetIn.decimals, assetOut: assetOut.decimals},
```

Comment:
Let's change the shape of SwapV2.getFixedInputSwapQuote and SwapV2.getFixedOutputSwapQuote
```suggestion
      assetIn,
      assetOut,
      pool: v2Pool?.info ?? null,
```